server/s3_log.cc

```diff
@@ -21,6 +21,18 @@
 #include "s3_log.h"
 #include "s3_option.h"
 
+enum {
+  KB = 1 << 10,
+};
+
+static thread_local char log_buffer[10*KB];
+char *__log_buff() {
+  return log_buffer;
+}
+size_t __log_buff_sz() {
+  return 10*KB - 1;
```

Comment:
If we are avoiding sizeof() for performance reasons, that's fine but the 10\*KB needs to be in a shared variable.  Having it hard-coded twice in the code is just asking for a future bug if someone changes it in one location and forgets to change it in the other.  I realize there are right next to each other so it might not happen but you could imagine that someone one day adds some other code between these two locations and that error becomes increasingly likely.

Reply:
defining a macro for 10*KB will be a good option, that macro can be used at both places

Reply:
Or the enum that defines KB could just do this:

enum {

TEN\_KB = 10 \* (1 \<\< 10),

};

Also, is 1 \<\< 10 unnecessarily complex?  Why not just 

TEN\_KB = 10 \* 1024

Reply:
`10*KB - 1` a compile-time constant and it will not hit performance.